crates/fuel-core/src/state/rocks_db.rs

```diff
@@ -706,6 +785,89 @@ where
             }
         }
     }
+
+    fn iter_store_keys(
```

Comment:
taken *mostly* as-is from the `iter_store` fn defn, but only uses keys and uses `_iter_all_keys` under the hood.

I have an implementation which abstracts the functionality of using key prefixes and start index, but it seems verbose and might not provide much value.